test_runner/src/main/kotlin/ftl/filter/TestFilters.kt

```diff
@@ -0,0 +1,181 @@
+package ftl.filter
+
+import com.linkedin.dex.parser.TestMethod
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.regex.Pattern
+
+interface ITestFilter {
```

Comment:
Maybe you want to match more the JUnit test filter? Meaning a method `shouldRun` and a `description`. It would be also possible to have a class that takes the description and a lambda.

Reply:
> Maybe you want to match more the JUnit test filter?

These aren't JUnit filters so I'm not sure it buys us much to match that API.
 
>  It would be also possible to have a class that takes the description and a lambda.

I'm interested in that idea. Do you think a class with a description/lambda is cleaner than the current interface and anonymous class approach?

If you want to try that refactor I'll take a look.

Reply:
> It would be also possible to have a class that takes the description and a lambda.

I pushed a new commit with that refactor.

Reply:
Updated to describe/shouldRun to match https://junit.org/junit4/javadoc/4.12/org/junit/runner/manipulation/Filter.html